html_css_dom/fetch/README.md

```diff
@@ -172,172 +179,106 @@ Now, we'll need to add our event listeners in our `index.js`:
-
-```js
-document.addEventListener('DOMContentLoaded', () => {
-    configureInputListeners()    
-})
-
-function configureInputListeners() {
-    getCountryInput().addEventListener('change', loadCountries)
-}
-
-function loadCountries() {
-    const searchTerm = getCountryInput().value
-    if (!searchTerm) { return }
-    fetch("https://restcountries.eu/rest/v2/name/" + searchTerm)
-        .then(response => {
-            return response.json()
-        })
-        .then(countries => {
-            removePreviousCards()
-            countries.forEach(country => {
-                createCardFromCountry(country)
-            })
-            return countries
-        })
-        .catch(error => {
-            console.log(error)
-        })
-}
-
-function getCountryInput() {
-    const countryInput = document.querySelector('#countryNameInput')
-    return countryInput
-}
-
-function getCountryContainer() {
-    const countryContainer = document.querySelector('#countryContainer')
-    return countryContainer
-}
-
-function removePreviousCards() {
-    // https://stackoverflow.com/questions/3955229/remove-all-child-elements-of-a-dom-node-in-javascript
-    const countryContainer = getCountryContainer()
-    console.log(countryContainer.firstChild)
-    while (countryContainer.firstChild) {
-        countryContainer.removeChild(countryContainer.firstChild)
-    }
-}
-
-function createCardFromCountry(country) {
-    let newCard = document.createElement('div')
-    newCard.className = "card"
-    newCard.style = 'width: 300px'
+We are using `fetch`, which returns a Promise that we that we can immediately attach a `.then` to to process the response once the promise is fulfilled. For now, we're just printing in the console the countries that come back to us. Test it out with different search terms and make sure that you're getting back data. 
 
-    let flagImage = document.createElement('img')
-    flagImage.src = `${country.flag}`
-    flagImage.style = 'width:200px'
+What happens when you search for the number 23? 
 
-    newCard.appendChild(flagImage)
+You'll notice that an error is thrown. The error says that `countries.forEach is not a function`. This is a little bit unexpected and undesired. I'd like for you to add a `debugger` on the line before `return response.json()` and one before the `forEach` then try again. While you're paused on the first debugger take a look at the Response object. Some things you may notice is that the Response object has the properties `status` and `ok`. Notice that the status is 404 (Not Found) and ok is `false`. Now play through the debugger. Now our code is frozen on the line before the `forEach`. This is undesired. We had a 404 status code but we still fulfilled our promise and moved on to the next chain. This is a NEGATIVE of `fetch`. 4xx and 5xx status codes will NOT throw errors with `fetch`. Instead we need to alter our fetch code like so: 
```

Comment:
👏 👏 👏  I love this, thank you for modeling debugging practices and helping fellows learn more about `fetch` through working with the code.